packages/svelte/src/compiler/phases/2-analyze/visitors/RegularElement.js

```diff
@@ -86,7 +86,10 @@ export function RegularElement(node, context) {
 		(attribute) => attribute.type === 'SpreadAttribute'
 	);
 
-	node.metadata.svg = is_svg(node.name);
+	node.metadata.svg =
+		is_svg(node.name) ||
+		(node.name === 'a' &&
+			context.path.some((n) => n.type === 'RegularElement' && n.name === 'svg'));
```

Comment:
What about a case like 

```svelte
<g>
	{#if true}
		<a href="/docs">
			<text x="20" y="40" class="small">{name}</text>
		</a>
	{/if}
</g>
```

But also what about this?

```svelte
<svg>
  <foreignObject>
    <div>
	{#if true}
		<a href="/docs">
			<text x="20" y="40" class="small">{name}</text>
		</a>
	{/if}
    </div>
  </foreignObject>
</svg>
```

-> I think it's better to traverse the path upwards, not downwards, and stop at the first regular element, and inherit its svg property

Reply:
Nice feedback, addressed.